msbuild/Xamarin.Mac.Tasks.Core/Tasks/ArchiveTaskBase.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.IO;
+using System.Diagnostics;
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+
+using Xamarin.MacDev;
+using Xamarin.MacDev.Tasks;
+
+namespace Xamarin.Mac.Tasks
+{
+    public abstract class ArchiveTaskBase : Xamarin.MacDev.Tasks.ArchiveTaskBase
+    {
+        public override bool Execute ()
+        {
+            var archiveDir = CreateArchiveDirectory ();
+
+            try {
+                var plist = PDictionary.FromFile (Path.Combine (AppBundleDir.ItemSpec, "Contents", "Info.plist"));
+                var productsDir = Path.Combine (archiveDir, "Products");
+
+                // Archive the Applications...
+                var appDestDir = Path.Combine (productsDir, "Applications", Path.GetFileName (AppBundleDir.ItemSpec));
+                Ditto (AppBundleDir.ItemSpec, appDestDir);
+
+                // Archive the dSYMs...
+                if (Directory.Exists (DSYMDir)) {
+                    var destDir = Path.Combine (archiveDir, "dSYMs", Path.GetFileName (DSYMDir));
+                    Ditto (DSYMDir, destDir);
+                }
+
+                // Generate an archive Info.plist
+                var arInfo = new PDictionary ();
+                var props = new PDictionary ();
+                props.Add ("ApplicationPath", new PString (string.Format ("Applications/{0}", Path.GetFileName (AppBundleDir.ItemSpec))));
+                props.Add ("CFBundleIdentifier", new PString (plist.GetCFBundleIdentifier ()));
+
+                var version = plist.GetCFBundleShortVersionString ();
+                var build = plist.GetCFBundleVersion ();
+                props.Add ("CFBundleShortVersionString", new PString (version ?? (build ?? "1.0")));
+                props.Add ("CFBundleVersion", new PString (build ?? "1.0"));
+                props.Add ("SigningIdentity", SigningKey);
+
+                arInfo.Add ("ArchiveVersion", new PNumber (2));
+                arInfo.Add ("CreationDate", Now);
+                arInfo.Add ("Name", new PString (plist.GetCFBundleName () ?? plist.GetCFBundleDisplayName ()));
+                arInfo.Add ("ApplicationProperties", props);
+
+                if (!string.IsNullOrEmpty (ProjectGuid))
+                    arInfo.Add ("ProjectGuid", new PString (ProjectGuid));
+
+                if (!string.IsNullOrEmpty (ProjectTypeGuids))
+                    arInfo.Add ("ProjectTypeGuids", new PString (ProjectTypeGuids));
+
+                if (!string.IsNullOrEmpty (SolutionPath)) {
+                    arInfo.Add ("SolutionName", new PString (Path.GetFileNameWithoutExtension (SolutionPath)));
+                    arInfo.Add ("SolutionPath", new PString (SolutionPath));
+                }
+
+                if (!string.IsNullOrEmpty (InsightsApiKey))
```

Comment:
@wjk is this an Apple thing or something else? Who's key is it?

Reply:
I simply copied it from the iOS version. The targets file implies it's the Xamarin Insights API key. I don't think it's used anywhere, honestly.